### PR TITLE
build: Enforce github issue with TODO comments

### LINKED ISSUE #1307: Local lint rule to disallow `TODO:` statements without linking to an issue
Comments should be treated as code and ambiguous comments should be disallowed.

### PR DESCRIPTION
**Pull request by pmdartus**
## Details

* Add new internal linting rule: `lwc-internal/no-invalid-todo`. This new linting rule enforce the presence of a github issue or a GUS work item number right along with the `TODO`.
* Align alll the existing `TODO`s to the new syntax.

Fixes #1307

## Does this PR introduce breaking changes?

* ✅ `No, it does not introduce breaking changes.`

If yes, please describe the impact and migration path for existing applications.

## The PR fulfills these requirements:
* Have tests for the proposed changes been added? ✅
* Have you followed [these instructions](../CONTRIBUTING.md#-commit-message-conventions) to clearly describe the issue being fixed or feature enhanced? ✅

### CODE REVIEW

File: scripts/eslint-plugin/rules/no-invalid-todo.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+const TODO_REGEX = /\*?\s+(TODO|todo)/;
+const VALID_TODO_REGEX = /\*?\s+(TODO|todo) \[((#\d+)|(\S+\/\S+#\d+)|(W-\d+))\]:/;
+
+/**
+ * This eslint rule checks that all the TODO comments contains a reference to a git issue or a GUS
+ * bug.
+ *
+ * Valid comment formats:
+ *   - `// TODO [#1234]: This is a todo`
+ *   - `// TODO [salesforce/observable-membrane#1234]: This is a todo`
```
Comment (apapko) [inline scripts/eslint-plugin/rules/no-invalid-todo.js]: Should we enforce git references to always contain the full link? This is relevant especially when the Todos are related to the work in other repos.

File: scripts/eslint-plugin/rules/no-invalid-todo.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            const { value } = comment;
+
+            const isTodo = value.match(TODO_REGEX);
+            const isValidTodo = value.match(VALID_TODO_REGEX);
+
+            if (isTodo && !isValidTodo) {
+                context.report({
+                    node: comment,
+                    message:
+                        'Invalid TODO comment format, the git issue reference is probably missing.',
```
Comment (apapko) [inline scripts/eslint-plugin/rules/no-invalid-todo.js]: Should we state that the gus is valid as well? eg: `Invalid TODO comment format. Please augment the comment with a git or gus issue number`.

File: scripts/eslint-plugin/rules/no-invalid-todo.js (hunk 1 above)
Comment (pmdartus) [reply scripts/eslint-plugin/rules/no-invalid-todo.js]: The `org/repo#issue` is enough to reference the git issue. Adding the entire URL will make the comment really long an unpleasant to read.

File: scripts/eslint-plugin/rules/no-invalid-todo.js (hunk 2 above)
Comment (pmdartus) [reply scripts/eslint-plugin/rules/no-invalid-todo.js]: IMO, we should remove GUS work item reference over time, that why the error message don't mention GUS work items.

File: scripts/eslint-plugin/rules/no-invalid-todo.js (hunk 1 above)
Comment (apapko) [reply scripts/eslint-plugin/rules/no-invalid-todo.js]: that is fine. What i'm asking is to disallow [#1234] format and to instead always ask for the org/repo#issue

File: scripts/eslint-plugin/rules/no-invalid-todo.js (hunk 2 above)
Comment (apapko) [reply scripts/eslint-plugin/rules/no-invalid-todo.js]: Agree, git issue has a label to create a gus item anyways.

File: scripts/eslint-plugin/rules/no-invalid-todo.js (hunk 1 above)
Comment (pmdartus) [reply scripts/eslint-plugin/rules/no-invalid-todo.js]: I see what you mean, I would like to keep the `org` and `repo` implicit if not necessary. 99% of the `TODO`s are referencing the `salesforce/lwc` repo.

Comment (apapko) [reply scripts/eslint-plugin/rules/no-invalid-todo.js]: Sounds good.

**Pull request merged**